src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java

```diff
@@ -116,16 +120,40 @@ private List<UnknownWord> selectWords(UUID userId, int selectSize) throws Except
             selectSize = initialSelectSize - selectedWords.size();
             confidences.clear();
             confidences.addAll(Arrays.asList(ConfidenceEnum.HIGH, ConfidenceEnum.HIGHEST));
-            
-            selectedWords.addAll(unknownWordRepository.findRandomByOwnerListUserIdAndOwnerListIsActiveAndConfidence(
-                userId, true, confidences, selectSize));
+
+
+            if (selectedWords.size() == initialSelectSize) {
+                selectedWords.addAll(unknownWordRepository.findRandomByOwnerListUserIdAndOwnerListIsActiveAndConfidence(
+                        userId, true, confidences, selectSize));
+                return selectedWords;
+            }
+
+            log.error("Defaulting to the predefined word list, as selected words size is not equal to initial select size, " +
+                    "selectedWords size: {}, initialSelectSize: {}", selectedWords.size(), initialSelectSize);
+
+            QPredefinedWordList predefinedWordList = readPredefinedWordListFromYamlFile(DEFAULT_WORD_LIST_FILE);
+
+            predefinedWordList.getWords().forEach(word -> {
+                if (selectedWords.size() < initialSelectSize) {
+                    UnknownWord unknownWord = UnknownWord.builder()
+                            .word(word)
+                            .confidence(ConfidenceEnum.LOWEST)
+                            .ownerList(null)
+                            .build();
+
+                    selectedWords.add(unknownWord);
+                }
+            });
 
             if (selectedWords.size() != initialSelectSize) {
                 log.error("Error in selecting new words, selected words size is not equal to initial select size, " +
                         "selectedWords size: {}, initialSelectSize: {}", selectedWords.size(), initialSelectSize);
-                throw new NotFoundException("Error in selecting new words, you need to add more unknown words to your word bank");
+                throw new SomethingWentWrongException("Error in selecting new words, you need to add more unknown words to your word bank");
```

Comment:
Just `You need to add more unknown words to your word bank` could be more user friendly if we are displaying this to the user. You can mark as resolved directly if not.